.github/cargo_local_patch.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+import toml
+import os
+
+def modify_dependency(details, new_path):
+    if 'path' not in details:
+        details['path'] = new_path
+        details.pop('version', None)
+        details.pop('git', None)
+        details.pop('rev', None)
+        details.pop('branch', None)
```

Comment:
It took me a minute to understand this one, and was unsure what it should do it there is already a path.
```suggestion
# Modify the dict details in-place, replacing version and git references with the given path.
def modify_dependency(details, new_path):
    if 'path' not in details:
        details['path'] = new_path
        details.pop('version', None)
        details.pop('git', None)
        details.pop('rev', None)
        details.pop('branch', None)
    else:
        raise ValueError(f"dependency has existing path: {details['path']}")
```

Reply:
I don't think this is what we should do, as typically the external repo could have intra-workspace dependencies, i.e.:
```
risc0-build-ethereum = { version = "0.8.0-alpha.1", default-features = false, path = "build" }
risc0-ethereum-contracts = { version = "0.8.0-alpha.1", default-features = false, path = "contracts" }
risc0-ethereum-relay = { version = "0.8.0-alpha.1", default-features = false, path = "relay" }
risc0-forge-ffi = { version = "0.8.0-alpha.1", default-features = false, path = "ffi" }
```

and we want to keep those

Reply:
well actually we could overwrite those as well, and they will just point to the same directory practically speaking

Reply:
Changing that was creating issues between risc0-ethereum and bonsai-foundry-template. I ended up checking if the path is a subdirectory of the specific repo, and if so it does not need to be patched